### PR TITLE
python-clock solution

### PR DESCRIPTION
**Pull request by cmdellinger**
solution to exercism.io clock exercise in python

cc @barberj

### CODE REVIEW

File: python/clock/clock.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+## ------
+## class:       Clock
+## purpose:     create a 24 hour clock that can add or subract minutes
+## ------
+
+class Clock:
+
+    # helper functions
+    def two_dig_str(self, x):
```
Comment (barberj) [inline python/clock/clock.py]: do you want your helper methods to be public?

File: python/clock/clock.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        set_min = minutes%60
+        if set_min < 0:
+            set_min += 60
+
+        return self.two_dig_str(set_hr) + ":" + self.two_dig_str(set_min)
+    
+    # init and call behavior
+    def __init__(self, hours, minutes):
+        self.time = self.clock_time(hours, minutes)
```
Comment (barberj) [inline python/clock/clock.py]: did you consider initializing and using a tuple for state? Would lend itself easier to manipulation in `add` and `subtract`

File: python/clock/clock.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    def __eq__(self, other):
+        return self.time == other.time
+
+    def __ne__(self, other):
+        return self.time != other.time
+
+    # methods
+    def add(self, minutes):
+        self.time = self.clock_time(int(self.time[0:2]), int(self.time[3:]) + minutes)
+        return self.time
```
Comment (barberj) [inline python/clock/clock.py]: the test shows passing the results of `add` and `subtract` to `str`. Would you expect the result of `add` to be a `str`?

**Comment by cmdellinger** [comment]
I refactored the clock class.

cc @barberj

File: python/clock/clock.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        return self.time != other.time
+
+    # methods
+    def add(self, minutes): # -> String (XX:XX)
+        """ add minutes to time and return new 24 hour clock string in form XX:XX """
+        self.time[1] += minutes
+        self.time = self.__clock_time(self.time)
+        return self.__clock_str(self.time)
+    
+    def subtract(self, minutes): # -> String (XX:XX)
```
Comment (barberj) [inline python/clock/clock.py]: isn't subtract really just adding a negative number?

File: python/clock/clock.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        """ subtract minutes from time and return new 24 hour clock string in form XX:XX """
+        self.time[1] -= minutes
+        self.time = self.__clock_time(self.time)
+        return self.__clock_str(self.time)
```
Comment (barberj) [inline python/clock/clock.py]: Why are you returning a str representation in add and subtract?

**Review by barberj — CHANGES_REQUESTED** [review]

**Comment by cmdellinger** [comment]
@barberj 
I updated based on your comments. The methods now return a Clock object and subtract just calls add.

**Review by barberj — APPROVED** [review]

**Comment by barberj** [comment]
👍

**Pull request merged**